### ISSUE TITLE
Entity Data Model Wizard is unable to load DLL 'SQLite.Interop.dll'

### ISSUE THREAD

**Issue by Brezelmann** [ErikEJ/SqlCeToolbox#834]
I am currently developing a class library that should interface with SQLite and Entity Framework. I therefore followed the guide given here: https://github.com/ErikEJ/SqlCeToolbox/wiki/EF6-workflow-with-SQLite-DDEX-provider 

After following the steps I wanted to run the Entity Data Model Wizard, selected Code First from database, on the next site new Connection, there SQLite Provider (Simple for EF6 by ErikEJ). I Entered a valid Connectionstring and hit Enter. I immediately receive the Error: "Unable to load DLL 'SQLite.Interop.dll'. The specified module could not be found. (Exception from HRESULT: 0x8007007E)". 
After that I checked if the dll's were missing, however I found them in my output path in a x64 or x86 Folder. 

Did I miss something or do I need to do something different?

### Steps to reproduce
Create a new Visual Studio Project and run the above guide.

### Further technical details
Version 4.7.669.0 - 1.490.146 downloads

SQL Server Compact 4.0 in GAC - Yes - 4.0.8876.1
SQL Server Compact 4.0 DbProvider - Yes

SQL Server Compact 4.0 DDEX provider - No
SQL Server Compact 4.0 Simple DDEX provider - Yes

SQL Server Compact 3.5 in GAC - No
SQL Server Compact 3.5 DbProvider - No

SQL Server Compact 3.5 DDEX provider - No

Sync Framework 2.1 SqlCe 3.5 provider - No

SQLite ADO.NET Provider included: 1.0.110.0
SQLite EF6 DbProvider in GAC - Yes
System.Data.SQLite DDEX provider - No
SQLite Simple DDEX provider - Yes

Database engine: SQLite

Visual Studio or SSMS version: 15.9.20

**Comment by ErikEJ** [comment]
You need to ensure that you are using the same System.data.sqlite version everywhere. Also look at similar closed issues in this repo.

**Comment by Brezelmann** [comment]
It should be the same installation everywhere. In my solution I have used Nuget to get the only available version of System.Data.SQLite (v. 1.0.112).
I downloaded and installed the latest build of the toolbox and used sqlite-netFx46-setup-bundle-x64-2015-1.0.112.0.exe to install Sqlite into the GAC.

my app.config looks like this:
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <configSections>
    <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" requirePermission="false" />
  </configSections>
  <entityFramework>
    <providers>
      <provider invariantName="System.Data.SQLite.EF6" type="System.Data.SQLite.EF6.SQLiteProviderServices, System.Data.SQLite.EF6" />
    </providers>
  </entityFramework>
  <system.data>
    <DbProviderFactories>
      <remove invariant="System.Data.SQLite.EF6" />
      <add name="SQLite Data Provider (Entity Framework 6)" invariant="System.Data.SQLite.EF6" description=".NET Framework Data Provider for SQLite (Entity Framework 6)" type="System.Data.SQLite.EF6.SQLiteProviderFactory, System.Data.SQLite.EF6" />
      <remove invariant="System.Data.SQLite" />
    </DbProviderFactories>
  </system.data>
</configuration>
```

I am using Entity Frramework 6.3.0 and Visual Studio Professional 15.9.20.

Do you see anything that I did wrong based on the wiki entry or how it should be done?

**Comment by ErikEJ** [comment]
Ah, I probably need to update to 112, or you can downgrade to .110

**Comment by Brezelmann** [comment]
Sounds like a good plan. What is your preferred Solution and how long would a change take?

**Comment by ErikEJ** [comment]
Well, the fastest solution for you would be to downgrade to .110 is you can.

**Comment by Brezelmann** [comment]
Let me rephrase my question from the post before: Do you have any intention of changing to 112 or would you change to 113 (as mentioned in the other open issue) directly when it is available? And if you'd do the update to 112 first how long do you think you need?

You are correct that the fastest solution for me would be to downgrade to 110 immediately and just use this. However as I am quite in the beginning of my development I could postpone the issues with my database for a little bit and write the Entity Framework Repositories and Bussiness services first before I come back to the database. That would save me the trouble of properly uninstalling all sqlite components for 112 from the project and the machine and reinstalling 110 afterwards. Thats why I asked if you would do the change to see what my next steps would be.

Anyways Thank you very much for your fast and great help in this matter.

**Comment by ErikEJ** [comment]
Soon 😊

**Comment by ErikEJ** [comment]
I also accept pull requests

**Comment by ErikEJ** [comment]
Closing as duplicate of #835

**Comment by ErikEJ** [comment]
Latest build updated to 1.0.112

**Comment by Brezelmann** [comment]
Thanks for the info. 
Just tested it again on a fresh classlibrary to check if the update works, however the error remains the same. The Project is currently created with: .Net Framework 4.7.2, the nuget Packages EntityFramework 6.3.0 and System.Data.SQLite 1.0.112. 
The app.config looks like this: 
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <configSections>
    <!-- For more information on Entity Framework configuration, visit http://go.microsoft.com/fwlink/?LinkID=237468 -->
    <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" requirePermission="false" />
  </configSections>
  <entityFramework>
    <providers>
      <provider invariantName="System.Data.SqlClient" type="System.Data.Entity.SqlServer.SqlProviderServices, EntityFramework.SqlServer" />
      <provider invariantName="System.Data.SQLite.EF6" type="System.Data.SQLite.EF6.SQLiteProviderServices, System.Data.SQLite.EF6" />
    </providers>
  </entityFramework>
  <system.data>
    <DbProviderFactories>
      <remove invariant="System.Data.SQLite.EF6" />
      <add name="SQLite Data Provider (Entity Framework 6)" invariant="System.Data.SQLite.EF6" description=".NET Framework Data Provider for SQLite (Entity Framework 6)" type="System.Data.SQLite.EF6.SQLiteProviderFactory, System.Data.SQLite.EF6" />
    <remove invariant="System.Data.SQLite" /><add name="SQLite Data Provider" invariant="System.Data.SQLite" description=".NET Framework Data Provider for SQLite" type="System.Data.SQLite.SQLiteFactory, System.Data.SQLite" /></DbProviderFactories>
  </system.data>
</configuration>
```

and the Info from the Tool ist this:
Version 4.7.670.0 - 1.490.146 downloads

SQL Server Compact 4.0 in GAC - Yes - 4.0.8876.1
SQL Server Compact 4.0 DbProvider - Yes

SQL Server Compact 4.0 DDEX provider - No
SQL Server Compact 4.0 Simple DDEX provider - Yes


SQL Server Compact 3.5 in GAC - No
SQL Server Compact 3.5 DbProvider - No

SQL Server Compact 3.5 DDEX provider - No


Sync Framework 2.1 SqlCe 3.5 provider - No


SQLite ADO.NET Provider included: 1.0.112.0
SQLite EF6 DbProvider in GAC - Yes
System.Data.SQLite DDEX provider - No
SQLite Simple DDEX provider - Yes

I also checked the output folder and I can find both x86 and x64 folders with a SQlite.Interop.dll inside. I then tried (for the sake of completeness) to copy it directly into the output folder, right next to my dll (didn't work, same error) and after that right next to the database file, that lies in the root of my solution directory. 
For the sake of completeness this is the error that I always got: 
![Error](https://user-images.githubusercontent.com/28568270/75518349-f7ac8580-5a00-11ea-9f85-1e27839bb8ab.PNG)

However now the Versions are all the same. Meaning the provider is on 112, the sqlite instance is on 112 and everything else should be the same as in your guide (correct me if I am wrong)
Any other suggestions?
Could a clean install of Windows/Visual Studio help (e.g. by ensuring there are no strange entries in any config files)?

**Comment by ErikEJ** [comment]
Have you checked the version in your machine config - basically gone through the wiki post sentence by sentence?

**Comment by Brezelmann** [comment]
I'll just post the "relevant" part of the machine.config here. so that you can see it as well. The version is 112 as far as I can tell.
Here the 32bit machine.config from C:\\Windows\Microsoft.Net\Framework\... :
```
<system.data>
        <DbProviderFactories>
            <add name="Microsoft SQL Server Compact Data Provider 4.0" invariant="System.Data.SqlServerCe.4.0"
                description=".NET Framework Data Provider for Microsoft SQL Server Compact"
                type="System.Data.SqlServerCe.SqlCeProviderFactory, System.Data.SqlServerCe, Version=4.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91" />
            <add name="SQLite Data Provider" invariant="System.Data.SQLite.EF6"
                description=".NET Framework Data Provider for SQLite" type="System.Data.SQLite.EF6.SQLiteProviderFactory, System.Data.SQLite.EF6, Version=1.0.112.0, Culture=neutral, PublicKeyToken=db937bc2d44ff139" />
            <add name="dotConnect for SQLite" invariant="Devart.Data.SQLite"
                description="Devart dotConnect for SQLite" type="Devart.Data.SQLite.SQLiteProviderFactory, Devart.Data.SQLite, Version=5.15.1583.0, Culture=neutral, PublicKeyToken=09af7300eec23701" />
        </DbProviderFactories>
    </system.data>
```

and here the 64bit:
```
<system.data>
        <DbProviderFactories>
            <add name="Microsoft SQL Server Compact Data Provider 4.0" invariant="System.Data.SqlServerCe.4.0"
                description=".NET Framework Data Provider for Microsoft SQL Server Compact"
                type="System.Data.SqlServerCe.SqlCeProviderFactory, System.Data.SqlServerCe, Version=4.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91" />
            <add name="SQLite Data Provider" invariant="System.Data.SQLite.EF6"
                description=".NET Framework Data Provider for SQLite" type="System.Data.SQLite.EF6.SQLiteProviderFactory, System.Data.SQLite.EF6, Version=1.0.112.0, Culture=neutral, PublicKeyToken=db937bc2d44ff139" />
            <add name="dotConnect for SQLite" invariant="Devart.Data.SQLite"
                description="Devart dotConnect for SQLite" type="Devart.Data.SQLite.SQLiteProviderFactory, Devart.Data.SQLite, Version=5.15.1583.0, Culture=neutral, PublicKeyToken=09af7300eec23701" />
        </DbProviderFactories>
    </system.data>
```

I've (hopefully 😄 ) followed the wiki post sentence by sentence. I've done the following in this order:

1. Installed the toolbox from the link https://github.com/ErikEJ/SqlCeToolbox/wiki/Release-notes
2. Installed this File: sqlite-netFx46-setup-bundle-x64-2015-1.0.112.0.exe doing a full install and    checking gac (I didn't get an option to install components for 2015, probably because I am running 2017)
3. Restarted Windows and therefore VS
4. Checked the about Page of the tool in VS (See my first post for that)
5. Manually checked the GAC anyway (looked like the one above except for the Devart part)
6. Added provider line to app.config
7. installed System.Data.SQLite with Nuget (didn't use Package manager console)
8. Build the project
9. Checked the app.config (looked exactly like the one above and in your example)
10.  Wanted to add a ADO.Net Data Model in the root of my classlibrary.
10.1 Selected Code First from Database.
10.2 Selected New Connection.
10.3 Selected SQLite Provider (Simple for EF6 by ErikEJ).
10.4 Added Connectionstring with the following format: Data Source="absolute Path to                  ConfigDb.sqlite".
10.5 clicked on Test Connection and received above error

Did I miss anything obvious?

**Comment by ErikEJ** [comment]
Have you tried using VS 2019?

**Comment by Brezelmann** [comment]
Currently not an option for my VS Professional 2017, but I could setup a vm with VS 2019 Community on it to test if that would help.

**Comment by ErikEJ** [comment]
At this point it probably would help

**Comment by Brezelmann** [comment]
Interesting...
Setup a Windows 10 Pro for Workstations (Version 1903 Buildnumber 18362.657) installed Mozilla, VS 2019 Community with enough tools to develop C# class libraries. After that I started following your wiki Post. I came to the point after the restart to check if SQLite is in the GAC. 
I started the toolbox by going from tools to the Toolbox, clicked on the about and received this:
Version 4.7.670.0 - 1,490,146 downloads

SQL Server Compact 4.0 in GAC - No
SQL Server Compact 4.0 DbProvider - No

SQL Server Compact 4.0 DDEX provider - No
SQL Server Compact 4.0 Simple DDEX provider - Yes


SQL Server Compact 3.5 in GAC - No
SQL Server Compact 3.5 DbProvider - No

SQL Server Compact 3.5 DDEX provider - No


Sync Framework 2.1 SqlCe 3.5 provider - No


SQLite ADO.NET Provider included: 1.0.109.0
SQLite EF6 DbProvider in GAC - No
System.Data.SQLite DDEX provider - No
SQLite Simple DDEX provider - Yes

First thing I saw was the the SQLite ADO.NET Provider received a downgrade from 1.0.112.0 to 1.0.109.0 and a second later I received the following error (without touching anything): 
![error](https://user-images.githubusercontent.com/28568270/75539965-6f87a980-5a1b-11ea-8376-cd79ed0fdc93.PNG)

As far as I could see the exception should either come from .Net being improperly installed or a windows problem. I therefore reinstalled .Net Framework 4.7.2 but the error appears again when I have the about page open. When I leave the Toolbox closed, but installed, the error doesn't appear.

After investigating that a little bit I went on to manipulate the machine.config like instructed. 
They looked like this on first opening machine.config (even though sqlite was instructed to add itself to GAC):
```
<system.data>
        <DbProviderFactories/>
    </system.data>
```

I replaced it with the line given in the wiki Post by copying it from there. Changed it for both 32 bit and 64 bit versions. I used the PMC to add System.Data.SQLite to the library and did a successfull build. package.config looks like this:
```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="EntityFramework" version="6.3.0" targetFramework="net472" />
  <package id="System.Data.SQLite" version="1.0.112.0" targetFramework="net472" />
  <package id="System.Data.SQLite.Core" version="1.0.112.0" targetFramework="net472" />
  <package id="System.Data.SQLite.EF6" version="1.0.112.0" targetFramework="net472" />
  <package id="System.Data.SQLite.Linq" version="1.0.112.0" targetFramework="net472" />
</packages>
```
and app.config like this:
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <configSections>
    <!-- For more information on Entity Framework configuration, visit http://go.microsoft.com/fwlink/?LinkID=237468 -->
    <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" requirePermission="false" />
  </configSections>
  <entityFramework>
    <providers>
      <provider invariantName="System.Data.SqlClient" type="System.Data.Entity.SqlServer.SqlProviderServices, EntityFramework.SqlServer" />
      <provider invariantName="System.Data.SQLite.EF6" type="System.Data.SQLite.EF6.SQLiteProviderServices, System.Data.SQLite.EF6" />
    </providers>
  </entityFramework>
  <system.data>
    <DbProviderFactories>
      <remove invariant="System.Data.SQLite.EF6" />
      <add name="SQLite Data Provider (Entity Framework 6)" invariant="System.Data.SQLite.EF6" description=".NET Framework Data Provider for SQLite (Entity Framework 6)" type="System.Data.SQLite.EF6.SQLiteProviderFactory, System.Data.SQLite.EF6" />
    <remove invariant="System.Data.SQLite" /><add name="SQLite Data Provider" invariant="System.Data.SQLite" description=".NET Framework Data Provider for SQLite" type="System.Data.SQLite.SQLiteFactory, System.Data.SQLite" /></DbProviderFactories>
  </system.data>
</configuration>
```

After that I started Entity Data Model Wizard an received the exception that VS is unable to load DLL 'SQLite.Interop.dll'. The output folder does contain the required x64 and x86 folders with SQLite.Interop.dll in it. I again pasted the dll into the root of my project, solution and output path but the error remains the same.

I used Visual Studio Community 16.4.5 for this Test. Hope you can figure out something with that 😄

**Comment by ErikEJ** [comment]
I am out of ideas, and this seems to be an issue with either the SQLite driver and or the EF Tools, none of which I support

**Comment by Brezelmann** [comment]
Thats unfortunate. I see if I can find something similar through google/github/stackoverflow and if there is nothing I'll probably raise an issue there and see if someone can figure something out. 

Thank you for using your time and skills to narrow down the possible error paths.

**Comment by AlexPerotti** [comment]
I got the same exact issue. I've trying anything suggested with no luck

**Comment by Brezelmann** [comment]
Well as far as I am concerned I tried to narrow it down further to see where the exact error lied. @ErikEJ stated it is either the driver from sqlite or the ef tools. 
I therefore tried using only one of the technologies to see if it still fails.
My setup was a barebone project using sqlite and NHibernate instead of EF and its tools and the results where that there was no error occuring. The connection was successfully made and I could use the sqlite db through NHibernate.
Therefore I am quite certain that the error lies in the Entity Framework implementation of either the tools or the adapter itself. But again as @ErikEJ already stated both cannot be fixed here, but only at the Entity Framework github. 
Try to raise an issue there and reference this one to see if someone over there can help you

**Comment by yubj** [comment]
In my case, when I followed the guide given here: [https://github.com/ErikEJ/SqlCeToolbox/wiki/EF6-workflow-with-SQLite-DDEX-provider](https://github.com/ErikEJ/SqlCeToolbox/wiki/EF6-workflow-with-SQLite-DDEX-provider) 
to intergate vs2019 with ef6 sqlite database first
using sqlite-netFx46-setup-bundle-x64-2015-1.0.112.0, I also get the error unable to load DLL 'SQLite.Interop.dll', but when I change to x86 SqlIte bundle, I get all things ok

**Comment by ErikEJ** [comment]
@yubj the guide States to use the x86 bundle!
